### PR TITLE
Add enabled variable and update docs

### PR DESCRIPTION
**Pull request by joshmyers**
## what
Add enabled variable and update docs

## why
So we can boolean creation of these resources from the caller module

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -53,11 +61,11 @@ resource "aws_security_group" "default" {
 
 module "dns" {
   source    = "git::https://github.com/cloudposse/terraform-aws-route53-cluster-hostname.git?ref=tags/0.2.5"
-  enabled   = "${length(var.zone_id) > 0 ? "true" : "false"}"
+  enabled   = "${var.enabled}"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  enabled   = "${local.enabled && length(var.zone_id) > 0 ? "true" : "false"}"
```

We do it in many modules.
Don't create the DNS record if zone ID is not provided

File: outputs.tf [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
 output "dns_name" {
-  value       = "${aws_efs_file_system.default.dns_name}"
+  value       = "${local.enabled ? local.dns_name : "" }"
   description = "DNS name"
 }
 
 output "mount_target_dns_names" {
-  value       = ["${aws_efs_mount_target.default.*.dns_name}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.dns_name : "" }"]
   description = "List of DNS names for the given subnet/AZ per documented convention."
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  description = "List of DNS names for the given subnet/AZ per documented convention"
```

File: outputs.tf [hunk 3]
```diff
@@ -1,39 +1,39 @@
...
 
 output "dns_name" {
-  value       = "${aws_efs_file_system.default.dns_name}"
+  value       = "${local.enabled ? local.dns_name : "" }"
   description = "DNS name"
 }
 
 output "mount_target_dns_names" {
-  value       = ["${aws_efs_mount_target.default.*.dns_name}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.dns_name : "" }"]
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  value       = ["${local.enabled ? aws_efs_mount_target.default.*.dns_name : list("") }"]
```

File: outputs.tf [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
   description = "List of DNS names for the given subnet/AZ per documented convention."
 }
 
 output "mount_target_ids" {
-  value       = ["${aws_efs_mount_target.default.*.id}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.id : "" }"]
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  value       = ["${local.enabled ? aws_efs_mount_target.default.*.id : list("") }"]
```

File: outputs.tf [hunk 5]
```diff
@@ -1,39 +1,39 @@
...
   description = "List of IDs of the EFS mount targets (one per Availability Zone)"
 }
 
 output "mount_target_ips" {
-  value       = ["${aws_efs_mount_target.default.*.ip_address}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.ip_address : "" }"]
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  value       = ["${local.enabled ? aws_efs_mount_target.default.*.ip_address : list("") }"]
```

File: outputs.tf [hunk 6]
```diff
@@ -1,39 +1,39 @@
...
   description = "List of IPs of the EFS mount targets (one per Availability Zone)"
 }
 
 output "network_interface_ids" {
-  value       = ["${aws_efs_mount_target.default.*.network_interface_id}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.network_interface_id : "" }"]
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  value       = ["${local.enabled ? aws_efs_mount_target.default.*.network_interface_id : list("") }"]
```

File: main.tf [hunk 7]
```diff
@@ -1,6 +1,7 @@
 # Define composite variables for resources
 module "label" {
   source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.1"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.5"
```

**Comment by joshmyers** [comment]
@aknysh updated

File: outputs.tf [hunk 8]
```diff
@@ -1,39 +1,39 @@
...
 output "mount_target_ips" {
-  value       = ["${aws_efs_mount_target.default.*.ip_address}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.ip_address : list("") }"]
   description = "List of IPs of the EFS mount targets (one per Availability Zone)"
 }
 
 output "network_interface_ids" {
-  value       = ["${aws_efs_mount_target.default.*.network_interface_id}"]
+  value       = ["${local.enabled ? aws_efs_mount_target.default.*.network_interface_id : list("") }"]
   description = "The IDs of the network interface that Amazon EFS created when it created the mount target."
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  description = "The IDs of the network interface that Amazon EFS created when it created the mount target"
```

**Comment by DDSloan96** [comment]
What is the correct way to use the module now as it broke with this PR 

```${local.enabled ? aws_efs_mount_target.default.*.ip_address : list("") }
* module.efs.output.network_interface_ids: At column 3, line 1: conditional operator cannot be used with list values in:

${local.enabled ? aws_efs_mount_target.default.*.network_interface_id : list("") }
* module.efs.output.mount_target_ids: At column 3, line 1: conditional operator cannot be used with list values in:

${local.enabled ? aws_efs_mount_target.default.*.id : list("") }
* module.efs.output.mount_target_dns_names: At column 3, line 1: conditional operator cannot be used with list values in:

${local.enabled ? aws_efs_mount_target.default.*.dns_name : list("") }```

**Pull request merged**